local-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -52,16 +57,16 @@ spec = describe "End-to-end test using a mocked chain though" $ do
               withHydraNode 3 carolSk [aliceVk, bobVk] $ \n3 -> do
                 waitForNodesConnected [n1, n2, n3]
                 let contestationPeriod = 3 -- TODO: Should be part of init
-                sendRequest n1 "Init"
+                sendRequest n1 [aesonQQ|{"req": "init"}|]
                 waitForResponse 3 [n1, n2, n3] "ReadyToCommit [VerKeyMockDSIGN 10,VerKeyMockDSIGN 20,VerKeyMockDSIGN 30]"
-                sendRequest n1 "Commit (fromList [1])"
-                sendRequest n2 "Commit (fromList [2])"
-                sendRequest n3 "Commit (fromList [3])"
+                sendRequest n1 [aesonQQ|{"req": "commit", "args": [1] }|]
+                sendRequest n2 [aesonQQ|{"req": "commit", "args": [2] }|]
+                sendRequest n3 [aesonQQ|{"req": "commit", "args": [3] }|]
```

Comment:
Why do we need quasi quotes here?

Reply:
Because it's handy and readable for constructing JSON ?

Reply:
(if you have a better proposal, I am glad to hear it :no_mouth:)

Reply:
I have used "builders" for an `Aeson.Value` in the past, but have also seem full string quasi-quotation. This seems to be something in between from the looks of it. Personally I prefer to use the former as the compiler helps in finding bugs when creating the json AST.

The following should be equal, but I would be interested in seeing how the two approaches compare in a more complicated example:

```suggestion
                sendRequest n3 $ object [ "req" .= "commit", "args" .= [3] ]
```

Reply:
I would also favor builders, I don't see what QQuoters add here, it adds a dependency and compillation time for little value.

Reply:
And as the name implies, Aeson builders will lead us into test builders which will simplify the tests.

Reply:
>  I don't see what  QQuoters add here, it adds a dependency and compillation time for little value.

Well, actually, the quoter comes from the aeson library itself so it's not like it is adding much dependency. Plus, it's not a mere string QQ, it does parse the JSON underneath and provides compile-time errors for invalid JSON. The added value to me really being in the "what you see is what you get" kind of spirit. With builders and implicit conversions it's sometimes easy to miss something.